ui-tests/src/test/java/org/openmrs/reference/AddFormTest.java

```diff
@@ -17,23 +22,39 @@
  * Created by nata on 24.06.15.
  */
 public class AddFormTest extends TestBase {
-    private HomePage homePage;
-    private HeaderPage headerPage;
-    private ManageFormsPage manageForm;
-    private ClinicianFacingPatientDashboardPage patientDashboardPage;
+    private static String name = "newFormTest";
+	 private static String description = "description of new form";
+	 private static  String version = "1.2";
+
+	 private AdministrationPage administrationPage;
+	 private ManageHtmlFormsPage manageHtmlFormsPage;
+     private HtmlFormsPage htmlFormsPage;
+     private HomePage homePage;
+     private HeaderPage headerPage;
+     private ManageFormsPage manageForm;
+     private ClinicianFacingPatientDashboardPage patientDashboardPage;
+  
 
     @Before
     public void setUp() throws Exception {
         homePage = new HomePage(page);
-        assertPage(homePage);
+        administrationPage = new AdministrationPage(page);
         headerPage = new HeaderPage(driver);
         manageForm = new ManageFormsPage(driver);
         patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
     }
 
-    @Ignore //ignore due to moving forms functionality
     @Test
+    @Category(BuildTests.class)
     public void addFormTest() throws Exception {
+    	manageHtmlFormsPage =  homePage.goToAdministration().clickOnManageHtmlForms();
+    	 if (manageHtmlFormsPage.overrideAddButtonElement()){
+    	     manageHtmlFormsPage.clickOnNewHtmlForm();
+             HtmlFormsPage newHtmlForms = new HtmlFormsPage(manageHtmlFormsPage);
```

Comment:
For this unless we want to get our hands dirt with catching nullpointer exceptions from 
  htmlformsPage.createNewFormTest(name, description,version) method, because this method need to be instantiated, Instantiating it in the before method didnt work for me,any suggestion will be more appreciated  however this still works perfect and its more readable perhaps cc @kaweesi

Reply:
what's wrong that it fails in before, you need to fix the NPEs